### PR TITLE
Fixed #24220 - Allowed lazy objects for success_url

### PR DESCRIPTION
**Pull request by tricoder42**
See https://code.djangoproject.com/ticket/24220 for more info

### CODE REVIEW

**Comment by MarkusH** [comment]
buildbot, test this please.

File: django/views/generic/edit.py [hunk 1]
```diff
@@ -163,6 +163,7 @@ def get_success_url(self):
         Returns the supplied URL.
         """
         if self.success_url:
+            self.success_url = force_text(self.success_url)
```
Comment (timgraham) [inline django/views/generic/edit.py]: should we include a comment that `force_text()`should be removed at the end of the deprecation period?

Comment (tricoder42) [inline django/views/generic/edit.py]: Hmm, that's a good idea. However, I can't find tests with new format-style placeholders. Does `obj.format` works when `obj` is lazy object?

Comment (tricoder42) [inline django/views/generic/edit.py]: Yes, it works.

I tried:

```python
    url(r'^edit/author/(?P<pk>[0-9]+)/delete/interpolate_redirect/$',
        views.NaiveAuthorDelete.as_view(
            success_url=SimpleLazyObject(lambda: '/edit/authors/create/?deleted={id}'))),
```

and tests passed. I'll add a comment then.

**Comment by tricoder42** [comment]
Updated diff and commit message. Thank you for review!

**Pull request merged**